uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/sick-request.controller.js

```diff
@@ -6,25 +6,26 @@ define([
 ], function (_, controllers) {
   controllers.controller('SicknessRequestCtrl', SicknessRequestCtrl);
 
-  SicknessRequestCtrl.$inject = ['$log', '$q', 'api.optionGroup', 'parentCtrl'];
+  SicknessRequestCtrl.$inject = ['$controller', '$log', '$q', 'api.optionGroup', 'parentCtrl'];
 
-  function SicknessRequestCtrl ($log, $q, OptionGroup, parentCtrl) {
+  function SicknessRequestCtrl ($controller, $log, $q, OptionGroup, parentCtrl) {
     $log.debug('SicknessRequestCtrl');
-    var vm = parentCtrl;
+    // We need to extend parent controller with
+    // calculateBalanceChange() function from Leave Request Controller
+    $controller('LeaveRequestCtrl', { parentCtrl: parentCtrl });
 
-    vm.checkSubmitConditions = checkSubmitConditions;
-    vm.isChecked = isChecked;
-    vm.isDocumentInRequest = isDocumentInRequest;
-
-    vm.initChildController = initChildController;
+    parentCtrl.checkSubmitConditions = checkSubmitConditions;
+    parentCtrl.isChecked = isChecked;
+    parentCtrl.isDocumentInRequest = isDocumentInRequest;
+    parentCtrl.initChildController = initChildController;
 
     /**
      * Checks if submit button can be enabled for user and returns true if successful
      *
      * @return {Boolean}
      */
     function checkSubmitConditions () {
-      return vm._canCalculateChange() && vm.request.sickness_reason;
+      return parentCtrl.canCalculateChange() && !!parentCtrl.request.sickness_reason;
```

Comment:
i see this tendency to use `!!` a lot. Is it justified? if it worked before without it, why the change?

Reply:
"if it worked before without it" - If we notice some inconsistencies we should refactor. Here is the explanation why I did it.

The function tells it returns a boolean. `canCalculateChange()` returns boolean, all good here. However if `canCalculateChange()` is **true**, then it will check and return `request.sickness_reason` which is **not** a boolean.

![image](https://user-images.githubusercontent.com/3973243/33770797-cced4190-dc26-11e7-8660-81e425c37f83.png)

That makes `checkSubmitConditions` sometimes not return boolean, but a type of `vm.request.sickness_reason`.

Reply:
i think
```js
return !!(parentCtrl.canCalculateChange() && parentCtrl.request.sickness_reason)
```
is preferable, as you are really interested in casting to boolean the outcome of the whole condition, not of individual parts

Reply:
Done.